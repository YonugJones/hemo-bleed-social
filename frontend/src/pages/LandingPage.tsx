import { Link } from 'react-router-dom'

const LandingPage = () => {
  return (
    <main className='flex flex-col justify-center items-center h-screen'>
      <div className='flex items-center text-[var(--green)] gap-5 mb-5'>
        <img src='https://placehold.co/60' alt='Hemo Bleed Social Logo' />
        <div className='flex flex-col justify-center items-center'>
          <h1 className='jaro-logo text-6xl'>Hemo Bleed Social</h1>
          <h2>Live. Log. Learn.</h2>
        </div>
      </div>
      <div className='flex gap-5'>
        <Link to='/signup'>
          <button className='bg-[var(--green)] hover:bg-[var(--light-green)] text-white font-semibold py-2 px-4 rounded shadow cursor-pointer'>
            Create your account
          </button>
        </Link>
        <Link to='/login'>
          <button className='bg-[var(--green)] hover:bg-[var(--light-green)] text-white font-semibold py-2 px-4 rounded shadow cursor-pointer'>
            Log in
          </button>
        </Link>
      </div>
    </main>
  )
}

export default LandingPage
