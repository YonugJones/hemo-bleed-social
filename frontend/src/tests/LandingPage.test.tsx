import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { MemoryRouter, Routes, Route } from 'react-router-dom'
import LandingPage from '../components/LandingPage'

describe('LandingPage', () => {
  it('renders page snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <LandingPage />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders Hemo Bleed Social heading and slogan', () => {
    render(
      <MemoryRouter>
        <LandingPage />
      </MemoryRouter>
    )

    expect(
      screen.getByRole('heading', { name: /hemo bleed social/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /live. log. learn/i })
    ).toBeInTheDocument()
  })

  it('renders "Create your account" and "Log in" buttons', () => {
    render(
      <MemoryRouter>
        <LandingPage />
      </MemoryRouter>
    )

    expect(
      screen.getByRole('button', { name: /create your account/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /log in/i })).toBeInTheDocument()
  })

  it('navigate to signup when "Create your account" is clicked', async () => {
    const user = userEvent.setup()
    render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path='/' element={<LandingPage />} />
          <Route path='/signup' element={<div>Signup Page</div>} />
        </Routes>
      </MemoryRouter>
    )

    const signupButton = screen.getByRole('button', {
      name: /create your account/i,
    })
    await user.click(signupButton)
    expect(screen.getByText(/signup page/i)).toBeInTheDocument()
  })

  it('navigate to login when "Log in" is clicked', async () => {
    const user = userEvent.setup()
    render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path='/' element={<LandingPage />} />
          <Route path='/login' element={<div>Login Page</div>} />
        </Routes>
      </MemoryRouter>
    )

    const loginButton = screen.getByRole('button', { name: /log in/i })
    await user.click(loginButton)
    expect(screen.getByText(/login page/i)).toBeInTheDocument()
  })
})
