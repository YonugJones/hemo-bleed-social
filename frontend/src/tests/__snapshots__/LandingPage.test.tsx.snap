// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LandingPage > renders page snapshot 1`] = `
<div>
  <main
    class="flex flex-col justify-center items-center h-screen"
  >
    <div
      class="flex items-center text-[var(--green)] gap-5 mb-5"
    >
      <img
        alt="Hemo Bleed Social Logo"
        src="https://placehold.co/60"
      />
      <div
        class="flex flex-col justify-center items-center"
      >
        <h1
          class="jaro-logo text-6xl"
        >
          Hemo Bleed Social
        </h1>
        <h2>
          Live. Log. Learn.
        </h2>
      </div>
    </div>
    <div
      class="flex gap-5"
    >
      <a
        data-discover="true"
        href="/signup"
      >
        <button
          class="bg-[var(--green)] hover:bg-[var(--light-green)] text-white font-semibold py-2 px-4 rounded shadow cursor-pointer"
        >
          Create your account
        </button>
      </a>
      <a
        data-discover="true"
        href="/login"
      >
        <button
          class="bg-[var(--green)] hover:bg-[var(--light-green)] text-white font-semibold py-2 px-4 rounded shadow cursor-pointer"
        >
          Log in
        </button>
      </a>
    </div>
  </main>
</div>
`;
